forge/test/mlir/test_loss.py

```diff
@@ -62,3 +62,36 @@ def test_cross_entropy_loss(prediction_shape):
     torch_loss_out = torch_loss(prediction, target)
 
     assert torch.allclose(torch_loss_out, forge_loss_out[0], rtol=11e-3)
+
+
+@pytest.mark.parametrize(
+    "prediction_shape",
+    [
+        (2, 2),
+        (3, 5),
+        (32, 32),
+        (33, 127),
+        (128, 20),
+        (128, 128),
+    ],
+)
+def test_nll_loss(prediction_shape):
+    forge_loss = forge.op.loss.NLLLoss("nll_loss")
+    torch_loss = torch.nn.NLLLoss()
+
+    prediction = torch.randn(prediction_shape, requires_grad=True).to(torch.float32)
+    prediction_forge = forge.tensor.Tensor.create_from_torch(prediction)
+    target = torch.empty(prediction_shape[0], dtype=torch.long).random_(prediction_shape[-1])
+
+    # Because of the following error
+    # RuntimeError: TT_FATAL @ ../embedding_device_operation.cpp:28: weights.get_dtype() == DataType::BFLOAT16
```

Comment:
could you check whether this ties to an existing issue? Maybe mention the issue nr here if it does.